src/main/java/com/tripply/booking/service/Impl/RoomServiceImpl.java

```diff
@@ -130,6 +135,55 @@ public ResponseModel<Room> getRoomDetailsById(Long id) {
         return responseModel;
     }
 
+    @Override
+    public ResponseModel<RoomBookingResponse> bookRoom(UUID hotelId, RoomBookingRequest request) {
+        log.info("RoomService: method -> bookRoom() with id: {} started", hotelId);
+        UserProfile user = userProfileRepository.findById(UUID.fromString(request.getUserId())).orElseThrow(() -> new DataNotFoundException("Specified user details not found in our system!"));
+        Hotel hotel = hotelRepository.findById(hotelId).orElseThrow(() -> new DataNotFoundException("Specified hotel details not found in our system!"));
+        List<Integer> availableRooms = roomRepository.findBySpecialFilters(hotelId, request.getCategory(), request.getType());
+
+        availableRooms.subList(request.getRoomCount(), availableRooms.size());
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss");
+        LocalDateTime checkInTime = LocalDateTime.parse(request.getCheckInTime(), formatter);
```

Comment:
Create convertStringIntoDateTime() in utils/DateTimeUtils.java and use try catch to avoid exception in main flow.